Automattic/pocket-casts-ios
# Animation + style tweaks on Discover Categories

Fixes a problem with reloading the categories selector view when navigating back away from a selected category. This was a regression from a previous PR.

| Before | After |
| -- | -- |
| ![Before](https://github.com/Automattic/pocket-casts-ios/assets/3250/7289b7c2-8d0d-4345-8a5e-49418821780b) | ![CleanShot 2024-04-15 at 16 14 18](https://github.com/Automattic/pocket-casts-ios/assets/3250/9ec21fff-af32-4703-95fb-814dd22a8d5d) |

Adjusts the close button to be fully circular:

<img src="https://github.com/Automattic/pocket-casts-ios/assets/3250/2c33d07e-82c3-4fb7-817a-e3454e5310e6" width=300>

Adds maximum number of 3 lines to sponsored Podcasts:

<img src="https://github.com/Automattic/pocket-casts-ios/assets/3250/58065620-7123-4948-b512-f0ac45cc6220" width=300>

## To test

Build using `Staging` configuration

* Enable the `categoriesRedesign` feature flag.
* Force reload discover from the Developer menu
* Try tapping one of the categories in the selector
* Ensure the animations works properly - it shouldn't fade the selector out, it should animate the selected pill
* Tap the X button to close
* Ensure the animations works properly - it shouldn't fade the selector out, it should animate the selected pill

### Sponsored Podcast

* Navigate to the "Sports" category
* Ensure that the Sponsored Podcast on top is truncated to only 3 lines

## Checklist
